server/internal/server/vector_stores.go

```diff
@@ -0,0 +1,321 @@
+	if err != nil {
+		return nil, err
+	}
+
+	id, err := getCollectionID(req.VectorStoreId, true)
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := s.store.GetCollectionByCollectionID(userInfo.ProjectID, id)
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, status.Errorf(codes.NotFound, "collection %q not found", req.VectorStoreId)
+		}
+		return nil, status.Errorf(codes.Internal, "get collection: %s", err)
+	}
+	if c.ProjectID != userInfo.ProjectID {
+		return nil, status.Errorf(codes.PermissionDenied, "collection %q not found", req.VectorStoreId)
+	}
+
+	// update collection metadata
+	cms, err := s.store.ListCollectionMetadataByCollectionID(c.CollectionID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "list collection metadata: %s", err)
+	}
+	cur := make(map[string]*store.CollectionMetadata)
+	for _, cm := range cms {
+		cur[cm.Key] = cm
+	}
+	for k, v := range req.Metadata {
+		found, ok := cur[k]
+		if !ok {
+			cm := &store.CollectionMetadata{
+				CollectionID: c.CollectionID,
+				Key:          k,
+				Value:        v,
+			}
+			if err := s.store.CreateCollectionMetadata(cm); err != nil {
+				return nil, status.Errorf(codes.Internal, "create collection metadata: %s", err)
+			}
+		} else if found.Value != v {
+			found.Value = v
+			if err := s.store.UpdateCollectionMetadata(found); err != nil {
+				return nil, status.Errorf(codes.Internal, "update collection metadata: %s", err)
+			}
+		}
+	}
+	for _, cm := range cms {
+		if _, ok := req.Metadata[cm.Key]; !ok {
+			if err := s.store.DeleteCollectionMetadata(cm.CollectionID, cm.Key); err != nil {
+				return nil, status.Errorf(codes.Internal, "delete collection metadata: %s", err)
+			}
+		}
+	}
+	cms, err = s.store.ListCollectionMetadataByCollectionID(c.CollectionID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "list collection metadata: %s", err)
+	}
+
+	// update collection
+	if req.Name != "" && req.Name != c.Name {
+		if err := s.vstoreClient.UpdateVectorStoreName(ctx, c.Name, req.Name); err != nil {
+			return nil, status.Errorf(codes.Internal, "update vector store: %s", err)
+		}
+		c.Name = req.Name
+	}
+	if req.ExpiresAfter != nil {
+		c.Anchor = store.ExpiresAfterAnchor(req.ExpiresAfter.Anchor)
+		c.ExpiresAfterDays = req.ExpiresAfter.Days
+	}
+	if err := s.store.UpdateCollection(c); err != nil {
+		return nil, status.Errorf(codes.Internal, "update collection: %s", err)
+	}
+
+	return toVectorStoreProto(c, cms), nil
+}
+
+// DeleteVectorStore deletes a vector store.
+func (s *S) DeleteVectorStore(
+	ctx context.Context,
+	req *v1.DeleteVectorStoreRequest,
+) (*v1.DeleteVectorStoreResponse, error) {
+	userInfo, err := s.extractUserInfoFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	id, err := getCollectionID(req.VectorStoreId, true)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := s.vstoreClient.DeleteVectorStore(ctx, req.VectorStoreId); err != nil {
```

Comment:
How about handling a not-found error? We can gracefully handle it or return `NotFound`.